pallets/parachain-staking/src/set.rs

```diff
@@ -191,54 +219,71 @@ impl<T: Ord> OrderedSet<T> {
 		self.0.is_empty()
 	}
 
-	/// Convert the set to a vector.
-	pub fn into_vec(self) -> Vec<T> {
+	/// Convert the set to a bounded vector.
+	pub fn into_bounded_vec(self) -> BoundedVec<T, S> {
 		self.0
 	}
 
-	/// Sort from greatest to lowest
+	/// Sorts from greatest to lowest.
+	///
+	/// NOTE: BoundedVec does not implement DerefMut because it would allow for
+	/// unchecked extension of the inner vector. Thus, we have to work with a
+	/// clone unfortunately.
```

Comment:
i would put that into a comment in the function. Don't think that users of the function need to know that.
Oh and we clone the whole vector. Can we not [swap](https://doc.rust-lang.org/std/mem/fn.swap.html) the `self.0` with a `Vec::with_capacity(0).into()` thing and swap it back later? But i'm not 100% sure that it's worth it...

Reply:
I think it's worth it! What do you say?
```
let mut sorted_v: sp_std::vec::Vec<T> = sp_std::mem::replace(&mut self.0, BoundedVec::default()).into();
sorted_v.sort_by(|a, b| b.cmp(a));
self.0 = sorted_v.try_into().expect("Did not extend size of bounded vec");
```

Reply:
Implemented in 77b653b